packages/graphql/src/mutations/marketplace/withdrawDust.js

```diff
@@ -22,14 +23,36 @@ async function withdrawDust(_, data) {
 
   const currency = await currencies.get(data.currency)
 
-  const tx = Proxy.methods.transferToOwner(
-    currency.address,
-    contracts.web3.utils.toWei(data.amount, 'ether')
-  )
+  const weiValue = contracts.web3.utils.toWei(data.amount, 'ether')
+
+  let currencyAddress = currency.address
+  if (!currencyAddress) {
+    const contractToken = contracts.tokens.find(t => t.symbol === currency.code)
+    if (contractToken) {
+      currencyAddress = contractToken.id
+    }
+  }
+
+  let tx = Proxy.methods.transferToOwner(currencyAddress, weiValue)
+
+  if (currency.code !== 'ETH') {
+    // For ERC20
+    const txData = await tx.encodeABI()
+
+    tx = Proxy.methods.marketplaceExecute(
+      owner,
+      proxy,
+      txData,
+      currencyAddress,
+      weiValue
+    )
+
+    gas = gas + 100000
```

Comment:
100k looks like a magic number. Is there a project wide constant that we can use or at least add some kind of comment why 100k?

Reply:
@sparrowDom That probably isn't necessary. But since we make delegatecall using `marketplaceExecute` method for ERC20, I didn't want the transaction to fail due to low gas. Can lower that amount and add a comment

Reply:
@sparrowDom Added the comment https://github.com/OriginProtocol/origin/pull/3406/commits/25add49522629542a8dc0fc2081666ea91226ac1